api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -86,6 +100,7 @@ type NodePool struct {
 // +kubebuilder:validation:XValidation:rule="!has(oldSelf.arch) || has(self.arch)", message="Arch is required once set"
 // +kubebuilder:validation:XValidation:rule="self.arch != 'arm64' || has(self.platform.aws) || has(self.platform.azure) || has(self.platform.agent) || self.platform.type == 'None'", message="Setting Arch to arm64 is only supported for AWS, Azure, Agent and None"
 // +kubebuilder:validation:XValidation:rule="!has(self.replicas) || !has(self.autoScaling)", message="Both replicas or autoScaling should not be set"
+// +kubebuilder:validation:XValidation:rule="!has(self.platform.aws) || !has(self.platform.aws.imageType) || self.platform.aws.imageType != 'windows' || (self.arch == 'amd64' && self.platform.type == 'AWS')", message="ImageType 'windows' is only supported with arch 'amd64' and platform type 'AWS'"
```

Comment:
For 4.16 and above APIs, you can simplify this. Does HCP have a minimum supported API server greater than 4.14 yet?
```suggestion
// +kubebuilder:validation:XValidation:rule="self.?platform.aws.imageType.orValue('') != 'windows' || (self.arch == 'amd64' && self.platform.type == 'AWS')", message="ImageType 'windows' is only supported with arch 'amd64' and platform type 'AWS'"
```